test/test_ecr_utils.py

```diff
@@ -0,0 +1,44 @@
+import pytest
+from unittest import mock
+from dcicutils.ecr_utils import ECRUtils
+from dcicutils.docker_utils import DockerUtils
+
+
+REPO_URL = '123456789.dkr.ecr.us-east-2.amazonaws.com/cgap-mastertest'
+
+
+def mocked_describe_respositories():
+    """ Mocks the important info from the describe_repositories API call. """
+    return {'repositories': [{
+        'repositoryUri': REPO_URL,
+       }]}
+
+
+def mocked_ecr_login(*, username, password, registry):
+    return
+
+
+def test_ecr_utils_basic():
+    """ Tests something simple for now, more tests to be added later. """
+    cli = ECRUtils()  # default args ok
+    with mock.patch.object(cli.client, 'describe_repositories', mocked_describe_respositories):
+        url = cli.resolve_repository_uri()
+        assert url == REPO_URL
+
+
+def test_ecr_utils_workflow():
+    """ Tests URL + Login via Docker_cli"""
+    ecr_cli = ECRUtils()
+    docker_cli = DockerUtils()
+    with mock.patch.object(ecr_cli.client, 'describe_repositories', mocked_describe_respositories):
+        ecr_cli.resolve_repository_uri()
+        ecr_user, ecr_pass = 'dummy', 'dummy'  # XXX: integrated test for this mechanism
```

Comment:
Minimally I might not pass the same thing for user and pass so you can later check they are passed in correct order. If these values were taken from environment variables, you could use the `override_environ` (I think it's in `qa_utils`, but if not it's in `misc_utils`) to bind the values. But that would